desktop/notification/fdo.go

```diff
@@ -105,18 +131,40 @@ func mapHints(hints []Hint) map[string]dbus.Variant {
 	return result
 }
 
+func fdoPriority(priority Priority) uint8 {
+	switch priority {
+	case PriorityLow:
+		return 0
+	case PriorityNormal, PriorityHigh:
+		return 1
+	case PriorityUrgent:
+		return 2
+	default:
+		return 1 // default to normal
+	}
+}
+
 // CloseNotification closes a notification message with the given ID.
-func (srv *Server) CloseNotification(id ID) error {
-	call := srv.obj.Call(dBusInterfaceName+".CloseNotification", 0, id)
-	return call.Store()
+func (srv *fdoBackend) CloseNotification(id ID) error {
+	serverSideId, ok := srv.localToServerID[id]
+	if !ok {
+		return fmt.Errorf("unknown notification with id %q", id)
+	}
+	call := srv.obj.Call(dBusInterfaceName+".CloseNotification", 0, serverSideId)
+	if err := call.Store(); err != nil {
+		return err
+	}
+	delete(srv.localToServerID, id)
+	delete(srv.serverToLocalID, serverSideId)
```

Comment:
I'm working on close signal handling in the followup PR. I'd prefer to keep this code around because removing it would require changes (or removal) of ObserveNotifications and would affect a bunch of existing tests. Since this flow isn't triggered yet (until followup), would it be ok to revisit this aspect there?

Reply:
Ah, I think what I can do is remove IdleDuration() and relate lastRemove, which will also address the remark from @mardy about unused code. Done.